### PR TITLE
Detect negated content absent buffer 2224 v12.4

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6575

Describe changes:
- detect: unify multi-buffer code

#10422 with fixed typo between krb4 cname and sname leading to failed SV

Preliminary work for #10334

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 101 | 112 | 110.89% | 

Pipeline 18494

**Comment by victorjulien** [comment]
I'm missing explanations in the git messages.

File: src/detect-engine.c [hunk 1]
```diff
@@ -2211,6 +2226,45 @@ uint8_t DetectEngineInspectBufferGeneric(DetectEngineCtx *de_ctx, DetectEngineTh
...
+        const Signature *s, Flow *f, uint8_t flags, void *alstate, void *txv, uint64_t tx_id)
+{
+    uint32_t local_id = 0;
+    const DetectEngineTransforms *transforms = NULL;
+    if (!engine->mpm) {
+        transforms = engine->v2.transforms;
+    }
+
+    while (1) {
+        InspectionBuffer *buffer = (engine->v2.GetMultiData)(
```
Comment (victorjulien) [inline src/detect-engine.c:2251]: style: not needed to wrap `engine->v2.GetMultiData`?

Comment (catenacyber) [reply src/detect-engine.c:2251]: ok

**Comment by catenacyber** [comment]
> I'm missing explanations in the git messages.

Tried something in https://github.com/OISF/suricata/pull/10462

> In general, there is a lot going on here, so git message should reflect that.

Why do you say there is "a lot" going on ?
This is just meant to be code deduplication

**Comment by victorjulien** [comment]
> > I'm missing explanations in the git messages.
> 
> Tried something in #10462
> 
> > In general, there is a lot going on here, so git message should reflect that.
> 
> Why do you say there is "a lot" going on ? This is just meant to be code deduplication

New API calls. I think that is worth adding some explanation about?

**Comment by catenacyber** [comment]
> New API calls. I think that is worth adding some explanation about?

Ok, I talk about `DetectAppLayerMultiRegister` in the new PR's commit, which is the only new API call, right ?

**Pull request closed without merging**